### PR TITLE
Sara/mkt 577 post launch tasks redirects

### PR DESCRIPTION
**Pull request by s-santillan**
# Thanks for improving Semgrep Docs 😀

### Please ensure

~~- [ ] A subject matter expert (SME) reviews the content~~ not required
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs

Creates redirects and new slug structure for our docs following the launch of Semgrep Code.

### CODE REVIEW

**Comment by s-santillan** [comment]
Note that this **does not have the semgrep-oss folder** - please check updated report as to why. 

To summarize - for our slug structure, I am suggesting we don't use a `semgrep-oss` folder any more.

File: docs/semgrep-code/semgrep-pro-intro.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-slug: deepsemgrep-introduction
+slug: semgrep-pro-intro
```
[the remark this answers is not in the document]
Comment (s-santillan) [reply docs/semgrep-code/semgrep-pro-intro.md]: Done!

File: docusaurus.config.js [hunk 2]
```diff
@@ -255,9 +255,44 @@ module.exports = {
...
-          { from: "/deepsemgrep/", to: "/deepsemgrep/deepsemgrep-introduction/" },
-          { from: "/semgrep-app/integrations", to: "/semgrep-app/notifications/" },
-          { from: "/troubleshooting/gitlab-sast/", to: "/troubleshooting/semgrep-ci/" }
+
+          /* MAR 24 2023 POST LAUNCH REDIRECTS */
+
+          // Deepsemgrep
+          { from: "/deepsemgrep/"                         , to: "/semgrep-code/semgrep-pro-engine-intro/" }     ,
+          { from: "/deepsemgrep/deepsemgrep-introduction" , to: "/semgrep-code/semgrep-pro-engine-intro/" }     ,
+          { from: "/deepsemgrep/deepsemgrep-examples"     , to: "/semgrep-code/semgrep-pro-examples/" }  ,
```
Comment (adamkvitek) [inline docusaurus.config.js:264]: Change `/semgrep-code/semgrep-pro-examples/` to `/semgrep-code/semgrep-pro-engine-examples/`. (Spell the product name fully.)

File: docusaurus.config.js [hunk 3]
```diff
@@ -255,9 +255,44 @@ module.exports = {
...
+          { from: "/deepsemgrep/semgrep-pro-data-flow"    , to: "/semgrep-code/semgrep-pro-data-flow/" } ,
```
Comment (adamkvitek) [inline docusaurus.config.js:265]: Change `"/semgrep-code/semgrep-pro-data-flow/" } ,` to `"/semgrep-code/semgrep-pro-engine-data-flow/" } ,`. (Spell the product name fully.)

File: docs/semgrep-code/overview.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ---
-slug: /semgrep-code/getting-started-with-semgrep-code 
+slug: getting-started
```
Comment (adamkvitek) [inline docs/semgrep-code/overview.md:2]: Shouldn't the slug be `overview` to be the same as the file name?

**Review by adamkvitek — COMMENTED** [review]
@s-santillan looks great, I left just a few comments (we should spell Semgrep Pro Engine fully everywhere, not just in the intro document). I am still not sure why ~so many of our slugs are different from the file name (maybe we can work on them later as this is not fully related to this update)~. Only one slug is different.

**Review by adamkvitek — APPROVED** [review]
Also approving so you can merge whenever you need to.

File: docs/faq.md [hunk 5]
```diff
@@ -96,7 +96,7 @@ Both Semgrep and CodeQL use static analysis to find bugs, but there are a few di
 - Semgrep operates directly on source code, whereas CodeQL requires a buildable environment
 - Semgrep is LGPL-2.1 and free to run anywhere; CodeQL is not open source and you must pay to run it on any non-open-source code
 - Semgrep supports autofixes; CodeQL does not.
-- Semgrep focuses on speed and ease of use. Semgrep doesn’t require compiled code, it doesn’t yet provide some of the analysis features of CodeQL. Semgrep has [limited intraprocedural dataflow](/writing-rules/data-flow/data-flow-overview/) and the proprietary [DeepSemgrep](/deepsemgrep/) extension to Semgrep provides similar capabilities as CodeQL in terms of interprocedural dataflow analysis for a subset of supported languages.
+- Semgrep focuses on speed and ease of use. Semgrep doesn’t require compiled code, it doesn’t yet provide some of the analysis features of CodeQL. Semgrep has [limited intraprocedural dataflow](/writing-rules/data-flow/data-flow-overview/) and the proprietary [DeepSemgrep](/semgrep-code/semgrep-pro-engine-intro) extension to Semgrep provides similar capabilities as CodeQL in terms of interprocedural dataflow analysis for a subset of supported languages.
```
Comment (pabloest) [inline docs/faq.md:99]:
```suggestion
- Semgrep focuses on speed and ease of use. Semgrep doesn’t require compiled code, it doesn’t yet provide some of the analysis features of CodeQL. Semgrep has [limited intraprocedural dataflow](/writing-rules/data-flow/data-flow-overview/) and the proprietary [DeepSemgrep, now part of the Pro Engine](/semgrep-code/semgrep-pro-engine-intro) extension to Semgrep provides similar capabilities as CodeQL in terms of interprocedural dataflow analysis for a subset of supported languages.
```

**Review by pabloest — APPROVED** [review]

**Pull request merged**